### PR TITLE
[HttpClient] Fix handling thrown \Exception in \Generator in MockResponse

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by fancyweb** [comment]
Failure is related, I need to investigate.

**Comment by fancyweb** [comment]
@nicolas-grekas The failure is related to the PHP version. It seems that the destructor of `ErrorChunk` is not triggered in the same way on 7.2 and above, I don't really know what should I do here. Keep the current behavior (direct throw) on 7.2? Modify the test so that it doesn't rely on the destructor? Something else?

File: src/Symfony/Component/HttpClient/Tests/MockHttpClientTest.php [hunk 1]
```diff
@@ -167,7 +210,7 @@ protected function getHttpClient(string $testCase): HttpClientInterface
             case 'testResolve':
                 $responses[] = new MockResponse($body, ['response_headers' => $headers]);
                 $responses[] = new MockResponse($body, ['response_headers' => $headers]);
-                $responses[] = new MockResponse((function () { throw new \Exception('Fake connection timeout'); yield ''; })(), ['response_headers' => $headers]);
+                $responses[] = new MockResponse((function () { yield ''; })(), ['response_headers' => $headers]);
```
Comment (fancyweb) [inline src/Symfony/Component/HttpClient/Tests/MockHttpClientTest.php:213]: We don't need to throw for the test to be effective. Yielding after throwing "breaks" PHP 7.2.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @fancyweb.

**Pull request merged**